tests/phpunit/Fixtures/PhpParser/RecordedVisitor.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Infection\Tests\Fixtures\PhpParser;
+
+use PhpParser\NodeVisitorAbstract;
+
+final class RecordedVisitor extends NodeVisitorAbstract
+{
+    private $records;
+
+    public function __construct(array &$records)
+    {
+        $this->records = &$records;
```

Comment:
This one create a recursion in the tree https://3v4l.org/J751R

I can't right now come up with an idea how it can influence us. Memory leaks in tests? 

Is there a way to do it in another way?

Reply:
I'm going to check for a potential memory leak. That said note that this is only for testing purposes (to check that the visitors are called in the right order) and that this was already existing code (I moved it from a private method of the test case into a dedicated fixture class)

Reply:
Couldn't find any evidence of a memory leak due to this so I think it's ok. It's not like I like this, but it's what was there and I didn't find a way to get around it either